xarray/core/resample.py

```diff
@@ -304,31 +307,36 @@ def _interpolate(self, kind='linear'):
         new_x = self._full_index.values.astype('float')
 
         arrs = []
-        # Apply the interpolation to each DataArray in our original Dataset
-        for v in self._obj.data_vars:
-            da = self._obj[v].copy()
-
-            x = da[self._dim].astype('float')
-            y = da.values
-            axis = da.get_axis_num(self._dim)
 
-            f = interp1d(x, y, kind=kind, axis=axis, bounds_error=True,
-                         assume_sorted=True)
+        # Prepare coordinates by dropping non-dimension coordinates along the
+        # resampling dimension.
+        # note: the lower-level Variable API could be used to speed this up
+        coords = OrderedDict()
+        if self._obj.data_vars:
+            var = list(self._obj.data_vars.keys())[0]
+            da = self._obj[var].copy()
 
-            # drop any existing non-dimension coordinates along the resampling
-            # dimension
-            coords = OrderedDict()
             for k, v in da.coords.items():
                 # is the resampling dimension
                 if k == self._dim:
                     coords[self._dim] = self._full_index
                 # else, check if resampling dim is in coordinate dimensions
                 elif self._dim not in v.dims:
                     coords[k] = v
-            dims = da.dims
+
+        # Apply the interpolation to each DataArray in our original Dataset
+        for var in self._obj.data_vars:
+            da = self._obj[var].copy()
+
+            x = da[self._dim].astype('float')
+            y = da.values
+            axis = da.get_axis_num(self._dim)
+
+            f = interp1d(x, y, kind=kind, axis=axis, bounds_error=True,
+                         assume_sorted=True)
```

Comment:
Do we check somewhere earlier that `assume_sorted=True` is valid? (If not, we'll need to check `Index.is_monotonic_increasing`)

Reply:
Yup! That gets checked by the logic we already use to build `GroupBy` objects, which powers resampling under the hood. See [here](https://github.com/pydata/xarray/blob/master/xarray/core/groupby.py#L234). If you try to build a non-monotonic timeseries, this error will raise long before we do any calculations.

Reply:
Sounds good, maybe add a quick comment noting that we already checked this in `__init__`? Since I already forgot :)